tests/rptest/tests/data_transforms_test.py

```diff
@@ -0,0 +1,127 @@
+        self._ctx = test_context
+        self._rpk = RpkTool(self.redpanda)
+
+    def _enable_wasm(self):
+        """
+        Enable data transforms and restart the cluster.
+        """
+        self._rpk.cluster_config_set("data_transforms_enabled", True)
+        self.redpanda.rolling_restart_nodes(self.redpanda.nodes)
+
+    def _deploy_wasm(self, name: str):
+        """
+        Deploy a wasm transform and wait for all processors to be running.
+        """
+        [input_topic, output_topic] = self.topics
+
+        def do_deploy():
+            self._rpk.deploy_wasm(name, input_topic.name, output_topic.name)
+            return True
+
+        wait_until(
+            do_deploy,
+            timeout_sec=30,
+            backoff_sec=5,
+            err_msg=f"unable to deploy wasm transform {name}",
+            retry_on_exc=True,
+        )
+
+        def is_all_running():
+            transforms = self._rpk.list_wasm()
+            transform = next((x for x in transforms if x.name == name), None)
+            if not transform:
+                raise WasmException(f"missing transform {name} from report")
+            for partition_id in range(0, input_topic.partition_count):
+                processor = next(
+                    (p
+                     for p in transform.status if p.partition == partition_id),
+                    None)
+                if not processor:
+                    raise WasmException(
+                        f"missing processor {name}/{partition_id} from report")
+                if processor.status != "running":
+                    raise WasmException(
+                        f"processor {name}/{partition_id} is not running: {processor.status}"
+                    )
+            return True
+
+        wait_until(
+            is_all_running,
+            timeout_sec=30,
+            backoff_sec=1,
+            err_msg=f"wasm transform {name} did not reach steady state",
+            retry_on_exc=True,
+        )
+
+    def _delete_wasm(self, name: str):
+        """
+        Delete a wasm transform and wait for all processors to stop.
+        """
+        def do_deploy():
+            self._rpk.delete_wasm(name)
+            return True
+
+        wait_until(
+            do_deploy,
+            timeout_sec=30,
+            backoff_sec=5,
+            err_msg=f"unable to delete wasm transform {name}",
+            retry_on_exc=True,
+        )
+
+        def transform_is_gone():
+            transforms = self._rpk.list_wasm()
+            transform = next((x for x in transforms if x.name == name), None)
+            if transform:
+                raise WasmException(f"transform {name} still in report")
+            return True
+
+        wait_until(
+            transform_is_gone,
+            timeout_sec=30,
+            backoff_sec=1,
+            err_msg=f"wasm transform {name} was not fully shutdown",
+            retry_on_exc=True,
+        )
```

Comment:
question: Similar to above, did you want to retry a couple of times instead of immediately erroring if the transform is still present?

Reply:
This does retry 😄 `retry_on_exc=True,`